code/game/objects/items.dm

```diff
@@ -571,10 +662,12 @@ var/list/global/slot_flags_enumeration = list(
 	I.Blend(new /icon('icons/effects/blood.dmi', rgb(255,255,255)),ICON_ADD) //fills the icon_state with white (except where it's transparent)
 	I.Blend(new /icon('icons/effects/blood.dmi', "itemblood"),ICON_MULTIPLY) //adds blood and the remaining white areas become transparant
 
+
 	//not sure if this is worth it. It attaches the blood_overlay to every item of the same type if they don't have one already made.
 	for(var/obj/item/A in world)

```

Comment:
Any thoughts on how this stupidity could be fixed? Maybe by caching? Use index access instead of running a for() loop. Granted, it doesn't seem like it's run to the full extent every single time, but, hnrg. It could probably be avoided all-together in an elegant fashion.